llama-index-core/llama_index/core/agent/function_calling/step.py

```diff
@@ -336,10 +337,10 @@ def run_step(self, step: TaskStep, task: Task, **kwargs: Any) -> TaskStepOutput:
                     tools,
                     tool_call,
                     task.extra_state["new_memory"],
-                    task.extra_state["sources"],
+                    tool_outputs,
                     verbose=self._verbose,
                 )
-
+                task.extra_state["sources"].append(tool_outputs[-1])
```

Comment:
I only add the last one since exactly one tool call is made in the line above. Really we don't need to pass a `sources` _list_ to `_call_function`, it could just return the new source, but I decided to leave this function unchanged.

I could wait until after the for-each-tool-loop and `extend` using the whole list, but I guess `task.extra_state["sources"]` may be updated sooner when appending individually. I'm open to changing this though!

Reply:
ah ok, I see what's going on here actually, this makes sense!